README.md
web2010
=======

Old web graphical charter non-responsive still used (sometimes partially) by several applications or web sites at EPFL. 

Description
-----------

This EPFL project contains:
* EPFL global header (non responsive website)
* templates html
* css files (non responsive)
* js files
* EPFL logos
* NGINX config

Setup
-----

You need to have Node.js, then:

```bash
$ npm install
```

You can now build this :factory:.

Update global header
--------------------

Edit the following files:
* templates/fragments/header.fr.html
* templates/fragments/header.en.html
* templates/fragments/header-no-local-search.fr.html
* templates/fragments/header-no-local-search.en.html
* navigate.fr.shtml
* navigate.en.shtml

Change "headerversion" + "headerdate":
* trunk/templates/fragments/header.sig.html

Commit the changes.

Change in JavaScript or CSS
---------------------------

Original files need to be in the `src` folder.


Build / Run
-----------

```bash
$ make up
```

Test
----

Test to download fragments:
```bash
wget -P <dest> -N -nd -r -l 1 http://localhost:8080/templates/fragments/download-me.html
```

Tagging
-------

Check the changes since the last release and update `CHANGELOG.md` accordingly.
Bump the new version and date in `templates/fragments/header.sig.html` and `package.json`.

Then

```bash
git tag -a v<version> -m "Version <version>"
git push --tags
```

Deploy
------

You need to have access to wwp-test and wwp OpenShift namespaces, and be logged in to OpenShift.

Clone [wp-ops](https://github.com/epfl-si/wp-ops) repository and deploy web2010:

`./ansible/wpsible --tags "web2010" --prod`  
`./ansible/wpsible --tags "web2010.rebuild"`  
`./ansible/wpsible --tags "web2010.image.promote" --prod`  

Routing url
-----------

The routing of urls trough 'www.epfl.ch' between WordPress and web2010 is operated by Varnish (see Varnish config) since 15th of September 2022.  
Here are the white list of urls forwarded to web2010:
* `^/accessibility`
* `^/images/`
* `^/cgi-bin/csoldap`
* `^/css/`
* `^/errors/`
* `^/javascript-help`
* `^/img/`
* `^/js/`
* `^/navigate`
* `^/templates`

Test web2010 urls:
`python test_web2010_urls.py`


CHANGELOG.md
Changelog
=========

### v3.7.0 / 2023-06-19

  - Update Google Analytics (UA → GA4)

### v3.6.1 / 2023-01-11

  - Disable hadolint rule DL3018 (pin version)

### v3.6.0 / 2023-01-10

  - Add script to test web2010 urls (routing)
  - Simplify and update Dockerfile
  - Remove inaccessible files in image
  - Remove label.php
  - Improve README

### v3.5.0 / 2022-04-12

  - Containerize web2010 (with migration to NGINX instead of Apache)
    - Dockerfile
    - NGINX config
  - Remove Bower and don't install Grunt globally

### v3.4.0 / 2019-06-19

  - Clean redirect and config
  - Remove homepage related stuff

### v3.3.0 / 2019-03-18

  - Change EPFL logos
  - Keep cookieconsent to v3.0.4

### v3.2.1 / 2018-12-17

  - Fix redirect with /index.html

### v3.2.0 / 2018-08-28

  - Remove search filter
  - Remove search autocomplete
  - Fix links in header

### v3.1.1 / 2018-07-18

  - Add EPFL Middle East section
  - Update dev dependencies
  - Remove tools folder
  - Disable `package-lock.json`

### v3.1.0 / 2018-03-22

  - Add proxy pass for label.php

### v3.0.2 / 2018-03-15

  - Remove all proxy configs (apache)

### v3.0.1 / 2018-03-13

  - Gzip svg (apache config)

### v3.0.0 / 2018-01-09

  - Feature improve deployment
  - Set 5s cache on html
  - Add homepage_versions script
  - Remove email.php
  - Remove jpgraph (PHP)
  - Remove other old PHP scripts
  - Add removed logos in 04.2017 (still used today)
  - Replace epfl_small.png (color too dark in previous)

### v2.20 / 2017-11-22

  - Fix resolution of html lang attribute

### v2.19 / 2017-10-17

  - Set cookie (from Cookie Consent) on localhost and external domain
  - Remove protocols in header

### v2.18 / 2017-10-11

  - Add science and society link in header
  - Remove injection of CSS in Cookie Consent
  - Fix cookie consent background link


js/jscroller.js
/*
 * jScroller 0.4 - Autoscroller PlugIn for jQuery
 *
 * Copyright (c) 2007 Markus Bordihn (http://markusbordihn.de)
 * Dual licensed under the MIT (MIT-LICENSE.txt)
 * and GPL (GPL-LICENSE.txt) licenses.
 *
 * $Date: 2009-06-18 20:00:00 +0100 (Sat, 18 Jul 2009) $
 * $Rev: 0.4 $
 */

eval(function(p,a,c,k,e,r){e=function(c){return(c<a?'':e(parseInt(c/a)))+((c=c%a)>35?String.fromCharCode(c+29):c.toString(36))};if(!''.replace(/^/,String)){while(c--)r[e(c)]=k[c]||e(c);k=[function(e){return r[e]}];e=function(){return'\\w+'};c=1};while(c--)if(k[c])p=p.replace(new RegExp('\\b'+e(c)+'\\b','g'),k[c]);return p}('$2={Q:{R:"S 2 T w U",V:0.4,W:"X Y (Z://10.11)",12:"13 14 15"},5:{g:[],x:16,s:{6:/([0-9,.\\-]+)6/}},h:{m:0,y:0},17:j(a,b,c,d,e){3($(a).F&&$(b).F&&c&&d>=1){$(a).k({18:\'19\'});$(b).k({1a:\'1b\',7:0,8:0});3(e){$(b).1c(j(){$2.n($(b),1d)},j(){$2.n($(b),G)})}$2.5.g.1e({z:$(a),f:$(b),H:c,o:d,n:G})}},n:j(a,b){3(a&&I b!==\'J\'){w(t i K $2.5.g){3($2.5.g[i].f.L("M")===a.L("M")){$2.5.g[i].n=b}}}},p:j(){3($2.h.m===0&&$2.5.x>0){$2.h.m=l.1f($2.A,$2.5.x)}3(!$2.h.y){$(l).1g($2.N);$(l).O($2.p);$(l).1h($2.p);$(l).A($2.p);$(1i).1j($2.p);3($.1k.1l){l.O()}$2.h.y=1}},N:j(){3($2.h.m){l.1m($2.h.m);$2.h.m=0}},B:{6:j(a){t b=\'\';3(a){3(a.C($2.5.s.6)){3(I a.C($2.5.s.6)[1]!==\'J\'){b=a.C($2.5.s.6)[1]}}}1n b}},A:j(){w(t i K $2.5.g){3($2.5.g.1o(i)){t a=$2.5.g[i],7=P(($2.B.6(a.f.k(\'7\'))||0)),8=P(($2.B.6(a.f.k(\'8\'))||0)),D=a.z.q(),E=a.z.r(),q=a.f.q(),r=a.f.r();3(!a.n){1p(a.H){u\'1q\':3(8<=-1*q){8=D}a.f.k(\'8\',8-a.o+\'6\');v;u\'1r\':3(7>=E){7=-1*r}a.f.k(\'7\',7+a.o+\'6\');v;u\'7\':3(7<=-1*r){7=E}a.f.k(\'7\',7-a.o+\'6\');v;u\'1s\':3(8>=D){8=-1*q}a.f.k(\'8\',8+a.o+\'6\');v}}}}}};',62,91,'||jScroller|if||config|px|left|top|||||||child|obj|cache||function|css|window|timer|pause|speed|start|height|width|regExp|var|case|break|for|refresh|init|parent|scroll|get|match|min_height|min_width|length|false|direction|typeof|undefined|in|attr|id|stop|focus|Number|info|Name|ByRei|Plugin|jQuery|Version|Author|Markus|Bordihn|http|markusbordihn|de|Description|Next|Generation|Autoscroller|120|add|overflow|hidden|position|absolute|hover|true|push|setInterval|blur|resize|document|mousemove|browser|msie|clearInterval|return|hasOwnProperty|switch|up|right|down'.split('|'),0,{}))
